api/server.go

```diff
@@ -26,6 +28,11 @@ type Server struct {
 	apiServer         *http.Server
 	listener          net.Listener
 	requiredUserAgent string
+
+	// wg is used to block Close() from returning until Serve() has finished. A
+	// WaitGroup is used instead of a chan struct{} so that Close() can be called
+	// without necessarily calling Serve() first.
+	wg sync.WaitGroup
```

Comment:
I don't think this is necessary, you should be able to just call close after the server returns, right?